exec/cnex/extension.c

```diff
@@ -0,0 +1,237 @@
+#include "array.h"
+#include "cell.h"
+#include "exec.h"
+#include "dictionary.h"
+#include "module.h"
+#include "neonext.h"
+#include "nstring.h"
+#include "object.h"
+#include "stack.h"
+#include "util.h"
+
+extern TExecutor *g_executor;
+
+struct Ne_ParameterList *parameterlist_alloc(int n)
+{
+    Cell *c = cell_newCell();
+    c->type = cArray;
+    c->array = array_createArrayFromSize(n);
+    return (struct Ne_ParameterList *)c;
+}
+
+void parameterlist_free(struct Ne_ParameterList *params)
+{
+    cell_freeCell(((Cell*)params));
+}
+
+int parameterlist_get_size(const struct Ne_ParameterList *list)
+{
+    return (int)((Cell*)list)->array->size;
+}
+
+int parameterlist_check_types(const struct Ne_ParameterList *list, const char *types)
+{
+    // ToDo: Follow this with exec.cpp
+    list = list; // Unused parameter
+    types = types; // Unused parameter
+    //fprintf(stderr, "parameterlist_check_types(%p, %s)\n", list, types);
+    return 1;
+}
+
+const struct Ne_Cell *parameterlist_get_cell(const struct Ne_ParameterList *list, int i)
+{
+    // Note: The array elements are stored in reverse order, so compute the index from the end.
+    return (struct Ne_Cell*)&((Cell*)list)->array->data[(((Cell*)list)->array->size - i) - 1];
+}
+
+struct Ne_Cell *parameterlist_set_cell(struct Ne_ParameterList *list, int i)
+{
+    return (struct Ne_Cell*)cell_arrayIndexForWrite((Cell*)list, i);
```

Comment:
The parameters are stored in the same order as for `parameterlist_get_cell`, so you have to do the same subtraction.